homeassistant/components/modem_callerid/__init__.py

```diff
@@ -1 +1,46 @@
-"""The modem_callerid component."""
+"""The Modem Caller ID integration."""
+import logging
+
+from phone_modem import PhoneModem
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DATA_KEY_API, DOMAIN, EXCEPTIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Modem Caller ID from a config entry."""
+    device = entry.data[CONF_DEVICE]
+    try:
+        api = PhoneModem(device)
```

Comment:
```suggestion
    api = PhoneModem(device)
    try:
```

You can move this outside of the `try` block now that there aren't any side effects that can throw

Reply:
What if the modem is not present during a reboot or reload?